kpi-tracker/resources/schema.sql

```diff
@@ -0,0 +1,53 @@
+CREATE TABLE IF NOT EXISTS blocks (
+  id SERIAL8 PRIMARY KEY,
+  hash BYTEA NOT NULL UNIQUE,
+  timestamp INT8 NOT NULL,
+  height INT8 NOT NULL,
+  total_stake INT8 -- NULL means the block is NOT a payday block.
+);
+
+CREATE TABLE IF NOT EXISTS accounts (
+  id SERIAL8 PRIMARY KEY,
+  address BYTEA NOT NULL UNIQUE,
+  block INT8 NOT NULL REFERENCES blocks(id) ON DELETE RESTRICT ON UPDATE RESTRICT, -- To support time series output.
+  is_initial BOOLEAN NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS modules (
+  id SERIAL8 PRIMARY KEY,
+  ref BYTEA NOT NULL UNIQUE,
+  block INT8 NOT NULL REFERENCES blocks(id) ON DELETE RESTRICT ON UPDATE RESTRICT -- To support time series output.
+);
+
+CREATE TABLE IF NOT EXISTS contracts (
+  id SERIAL8 PRIMARY KEY,
+  index INT8 NOT NULL,
+  subindex INT8 NOT NULL,
+  module BYTEA NOT NULL,
+  block INT8 NOT NULL REFERENCES blocks(id) ON DELETE RESTRICT ON UPDATE RESTRICT, -- To support time series output.
+  UNIQUE (index, subindex)
+);
+
+CREATE TABLE IF NOT EXISTS transactions (
+  id SERIAL8 PRIMARY KEY,
+  hash BYTEA NOT NULL UNIQUE,
+  block INT8 NOT NULL REFERENCES blocks(id) ON DELETE RESTRICT ON UPDATE RESTRICT, -- To support time series output.
+  type INT8 -- NULL means the transaction was rejected
+);
+
+-- Keeps track of relations between accounts and transactions to support account activeness.
+CREATE TABLE IF NOT EXISTS accounts_transactions (
+  account INT8 REFERENCES accounts(id) ON DELETE RESTRICT ON UPDATE RESTRICT,
+  transaction INT8 REFERENCES transactions(id) ON DELETE RESTRICT ON UPDATE RESTRICT,
+  PRIMARY KEY (account, transaction) -- Ensures only unique rows can be inserted
```

Comment:
Shouldn't this just be a unique constraint?

Reply:
My thought was that by using primary key I got unique + not null. Is there any overhead in using primary key?